plugins/net_plugin/net_plugin.cpp

```diff
@@ -1994,18 +1995,30 @@ namespace eosio {
    }
    //-----------------------------------------------------------
 
-    sync_manager::sync_manager( uint32_t span, uint32_t sync_peer_limit, uint32_t min_blocks_distance )
+    sync_manager::sync_manager( uint32_t span, uint32_t sync_peer_limit, size_t max_reversible_blocks, uint32_t min_blocks_distance )
       :sync_known_lib_num( 0 )
       ,sync_last_requested_num( 0 )
       ,sync_next_expected_num( 1 )
       ,sync_source()
-      ,sync_req_span( span )
+      ,sync_fetch_span( span )
       ,sync_peer_limit( sync_peer_limit )
+      ,max_reversible_blocks(max_reversible_blocks)
       ,sync_state(in_sync)
       ,min_blocks_distance(min_blocks_distance)
    {
    }
 
+   uint32_t sync_manager::active_sync_fetch_span() const {
+      auto fork_db_size = my_impl->chain_plug->chain().fork_db_size();
+      auto reversible_remaining = max_reversible_blocks - fork_db_size;
```

Comment:
We shutdown when `fork_db_size() >= max_reversible_blocks`, so really we should have:
```suggestion
      auto reversible_remaining = max_reversible_blocks - fork_db_size - 1;
```
I think.

Reply:
Do we need to check `max_reversible_blocks > fork_db_size` before doing subtract?